src/tags/core.js

```diff
@@ -23,10 +23,18 @@ export const nullObj = {
     ctx.wrapScalars ? new Scalar(null) : null,
   default: true,
   tag: 'tag:yaml.org,2002:null',
+  sourceStr: null,
   test: /^(?:~|[Nn]ull|NULL)?$/,
-  resolve: () => null,
+  customTest: str => {
+    return str === nullOptions.nullStr ? [str] : null
+  },
```

Comment:
This `customTest` approach makes it far too easy to produce invalid YAML just by using a bad `nullStr` option. It might be fine to _output_  the bad YAML if that's what the user really wants, but that doesn't mean that we should just accept it as input. To handle that, they should modify the `test` regexp that their schema uses.

Reply:
Removed.